testcases/OpTestSTB.py

```diff
@@ -0,0 +1,623 @@
+            if i_str in l_line:
+                # Removes the time stamp ex: [    4.195376870,5]
+                l_group = re.search(BMC_CONST.RE_TIMESTAMP, l_line)
+                if l_group is not None:
+                    l_data = l_group.groups()[0]
+                    return l_data
+        # If string not found retrun None
+        return None
+
+    ##
+    # @brief This function finds given strings @i_enable or @i_disable
+    # from a line @i_line.
+    # @i_line contains required message line from OPAL message log
+    # @i_enable contains string to be found ex: on
+    # @i_disable contains string to be found ex: off
+    # Returns either enabled @i_enable or disabled @i_disable string
+    #
+    def status_finder(self, i_line, i_enable, i_disable):
+        #for mode in self.i_generator:
+        if i_enable in i_line.rstrip(BMC_CONST.NEWLINE):
+            return i_enable
+        else:
+            return i_disable
+
+    ##
+    # @brief This function finds whether Secure and Trusted boot disabled
+    # information is there in OPAL message log.
+    # Returns either FW_FAILED or FW_SUCCESS based on the current support.
+    #
+    def find_stb_support(self):
+        l_stb_not_support = self.string_get(self.l_data,
+                                            BMC_CONST.STB_NOT_SUPPORTED)
+        if l_stb_not_support:
+            print BMC_CONST.STB_NOT_SUPPORTED
+            print "No need to run further tests. Bailing out..."
+            return BMC_CONST.FW_FAILED
+        else:
+            print "Run further tests ..."
+            return BMC_CONST.FW_SUCCESS
+
+    ##
+    # @brief This function reads Security Switch Register (SSR)
+    # of hardware for Secure and Trusted boot mode state
+    # This function compiles skiboot and with the help of 'getscom' binary
+    # reads from SSR.
+    # Returns @l_res, which contains register value in integer
+    # If the SSR value not found/can't read returns '0'
+    #
+    def read_hw_stb_support(self):
+        # Check whether git and gcc commands are available on the host
+        self.cv_HOST.host_check_command("git")
+        self.cv_HOST.host_check_command("gcc")
+
+        # It will clone skiboot source repository
+        l_dir = BMC_CONST.CLONE_SKIBOOT_DIR
+        self.cv_HOST.host_clone_skiboot_source(l_dir)
+        # Compile the necessary tools xscom-utils and gard utility
+        self.cv_HOST.host_compile_xscom_utilities(l_dir)
+
+        # Reading Security Switch Register for Secure boot values
+        l_cmd = "cd %s/external/xscom-utils/; ./getscom pu 10005" % l_dir
+        # Expected values
+        # cd38000000000000 - Secure and Trusted mode Enabled
+        # 0000000000000000 - Secure and Trusted mode Disabled
+        l_res = self.cv_HOST.host_run_command(l_cmd)
+        l_res = "0x" + l_res.replace("\r\n","")
+        l_res = int(l_res, 16)
+        if l_res:
+            return l_res
+        else:
+            return "0"
```

Comment:
don't check for zero value, just return the l_res i.e what getscom returns. otherwise returning "0" is failing on a system which is having both S&TB modes are off.
  File "/home/pridhiviraj/new_optests/sastry/op-test-framework/testcases/OpTestSTB.py", line 181, in hw_sb_support
    l_sb_bv = l_hw_sb_value & BMC_CONST.SB_MASK
TypeError: unsupported operand type(s) for &: 'str' and 'long'


So just return l_res.